utils.js

```diff
@@ -449,51 +516,44 @@ const createMethods = async ({ apig, apiId, endpoints }) => {
   return endpoints
 }
 
-const createModel = async ({ apig, apiId, model }) => {
-  const params = {
-    'contentType': 'application/json',
-    name: model.title,
-    restApiId: apiId,
-    description: model.description || null,
-    schema: JSON.stringify(model, null, '\t')
-  }
-
-  try {
-    await apig.createModel(params).promise()
-  } catch(e) {
-    throw Error(e)
-  }
-}
-
-const createModels = async ({ apig, apiId, models }) => {
-  for (const model of models) {
-    // models must be created one at a time in case they reference eachother
-    await createModel({ apig, apiId, model })
-  }
-
-  return models
-}
-
 const createMethodResponse = ({ apig, apiId, endpoint }) => {
   const promises = []
 
-  if (!endpoint.responses) return []
-
-  for (const response of endpoint.responses) {
+  for (const response of (endpoint.responses || [])) {
     const params = {
       httpMethod: endpoint.method,
       resourceId: endpoint.id,
       restApiId: apiId,
-      statusCode: `${response.code}`
+      statusCode: `${response.code}`,
+      responseModels: {
+        'application/json': response.model || 'Empty'
+      },
+      responseParameters: response.headers ? Object.keys(response.headers).reduce((res, ele) => {
+        res[`method.response.header.${ele}`] = false
+        return res
+      }, {})
+      : {}
     }
 
-    if (response.model) {
-      params.responseModels = {
-        'application/json': response.model
-      }
-    }
+    const p = apig.putMethodResponse(params).promise()
+      .catch(e => {
+        if (e.code === 'ConflictException') {
+          const params = {
+            httpMethod: endpoint.method,
+            resourceId: endpoint.id,
+            restApiId: apiId,
+            statusCode: `${response.code}`,
+            patchOperations:[{
+              op: 'replace',
+              path: '/responseParameters'
```

Comment:
Kinda curious what this does.

Reply:
Yeah so aws has an update for all their create functions. Put is more like a create or update. So update you use json pointer to identify a field on a resource that you want to modify. Didn't make sense to me until I found this https://docs.aws.amazon.com/apigateway/api-reference/link-relation/model-update/